app/views/profiles/show.html.erb

```diff
@@ -1,21 +1,60 @@
 <div class="uk-height-viewport uk-responsive-width">
-	<div class="uk-container-center"><!-- uk-scrollable-box uk-height-viewport uk-responsive-width">-->
-		<div class="uk-panel uk-panel-box">
-			<h3 class="uk-panel-title">Adatlap - <%= @user.full_name %></h3>
-			<!--<nav class="uk-navbar uk-hidden-small uk-width" style="width:282px;">
-				<ul class="uk-navbar-nav">
-					<li class="uk-active"><a href="">Profil</a></li>
-					<li><a href="">Közösségi profil</a></li>
-					<li><a href="">Közösségi történet</a></li>
-				</ul>
-			</nav>
--->
-			<%= image_tag photo_path(@user.screen_name), class: "uk-thumbnail", alt: @user.screen_name %>
-			<h3 class="uk-h3"><%= @user.full_name %></h3>
-			<h4 class="uk-h4"><%= @user.email %></h4>
-			<h4 class="uk-h4"><%= @user.dormitory %> <%= @user.room %></h4>
-			<h4 class="uk-h4"><%= @user.cell_phone %></h4>
+  <div class="uk-container-center">
+    <div class="uk-panel uk-panel-box">
+      <h3 class="uk-panel-title">
+        Adatlap
+        <% if not @user.nickname.blank? %>
+          - <%= @user.nickname %>
+        <% end %>
+      </h3>
 
-		</div>
-	</div>
+      <%= image_tag photo_path(@user.screen_name), class: "uk-thumbnail", alt: @user.full_name %>
+      <h3 class="uk-h3"><%= @user.full_name %></h3>
+
+      <% if not @user.email.blank? %>
+        <h4 class="uk-h4"><%= @user.email %></h4>
+      <% end %>
+
+      <% if not @user.dormitory.blank? %>
+        <h4 class="uk-h4"><%= @user.dormitory + ' ' + @user.room %></h4>
```

Comment:
Please don't put logic in the view. You have several options to avoid this:

* Add a method on the model and call that here
* Add a helper method in `UsersHelper`
* Create a view model/presenter object which delegates to the underlying user but also add view specific methods. You can [roll your own](http://nithinbekal.com/posts/rails-presenters/) or just use a [gem](https://github.com/drapergem/draper). Using a presenter is a more object oriented solution and you can hide a lot of view related code in those presenter. You could potentially replace the whole `if` block with a single method call.

```ruby
# in your presenter
def room
  return if @user.dormitory.blank?
  room = [@user.dormitory, @user.room].join(' ')
  view_context.content_tag(:h4, room, class: 'uk-h4')
end
```

```erb
<%# in your view %>
<%= @user_presenter.room %>
```